metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -1825,6 +1826,10 @@ class MetalsLspService(
             Option(params.uri).map(_.toAbsolutePath)
           )
         }.asJavaObject
+      case ServerCommands.DiscoverMainClasses(unresolvedParams) =>
```

Comment:
Maybe I'm missing the context, but reading over the VS Code pr it wasn't clear. Isn't this essentially the same as

```
[Trace - 08:58:37 am] Received request 'workspace/executeCommand - (21)'
Params: {
  "command": "metals.debug-adapter-start",
  "arguments": [
    {
      "path": "file:///Users/ckipp/Documents/scala-workspace/minimal/minimal3/src/Main.scala",
      "runType": "runOrTestFile"
    }
  ]
}
```
Which already works to discover mains int he current file.  I find this a bit confusing because now we have multiple different ways to discover main classes. Couldn't the existing `runtType` here with `debug-adapter-start` be re-utilized to do this same thing?

Reply:
The issue is that the way we previously did this is by starting the session itself within Metals. Here we just want to get the command to run on the client. So we could potentially add a parameter (`returnShellCommand: true` or something along these lines), but then this flag would make the behaviour widely different. One would start app and return a address to connect to and the other would return the session object. I think we shouldn't to it in the same command

Reply:
Also, this only adds a single new command, while reusing all the same mechanism, getting to work it with a flag might end up with considerable more effort.

Reply:
Ahh ok ok, make sense. Took some more time to really understand it. Thanks for explaining!